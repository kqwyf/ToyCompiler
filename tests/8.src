int input(int a) {
    do; /* Do something */
}
int height;
int width;
input(height);
input(width);
/* Here we multiply the height and width
 * so that we get the area of the rectangle.
 */
result = height * width;
print(result);
