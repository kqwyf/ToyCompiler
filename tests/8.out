Token sequence:
int             < DATATYPE    , INT    >
input           < IDENTIFIER  , 1      >
(               < LEFTBRACKET ,        >
int             < DATATYPE    , INT    >
a               < IDENTIFIER  , 2      >
)               < RIGHTBRACKET,        >
{               < LEFTBRACE   ,        >
do              < DO          ,        >
;               < SEMICOLON   ,        >
/* ... */       < COMMENT     ,        >
}               < RIGHTBRACE  ,        >
int             < DATATYPE    , INT    >
height          < IDENTIFIER  , 3      >
;               < SEMICOLON   ,        >
int             < DATATYPE    , INT    >
width           < IDENTIFIER  , 4      >
;               < SEMICOLON   ,        >
input           < IDENTIFIER  , 1      >
(               < LEFTBRACKET ,        >
height          < IDENTIFIER  , 3      >
)               < RIGHTBRACKET,        >
;               < SEMICOLON   ,        >
input           < IDENTIFIER  , 1      >
(               < LEFTBRACKET ,        >
width           < IDENTIFIER  , 4      >
)               < RIGHTBRACKET,        >
;               < SEMICOLON   ,        >
/* ... */       < COMMENT     ,        >
result          < IDENTIFIER  , 5      >
=               < ASSIGN      ,        >
height          < IDENTIFIER  , 3      >
*               < MULTIPLY    ,        >
width           < IDENTIFIER  , 4      >
;               < SEMICOLON   ,        >
print           < IDENTIFIER  , 6      >
(               < LEFTBRACKET ,        >
result          < IDENTIFIER  , 5      >
)               < RIGHTBRACKET,        >
;               < SEMICOLON   ,        >

Symbol table:
1     input
2     a
3     height
4     width
5     result
6     print
